src/dupstring.c

```diff
@@ -0,0 +1,24 @@
+#include "dupstring.h"
+
+#include <string.h>
+
+#if defined(HAVE_STRNDUP) && HAVE_STRNDUP
+
+char *dupstring(const char *source, size_t maxlen) {
+  return strndup(source, maxlen);
+}
+
+#else /* Fallback implementation */
+#include <stdlib.h>
+
+char *dupstring(const char *source, size_t maxlen) {
+  size_t dest_size = strnlen(source, maxlen);
+  char *dest = malloc(dest_size + 1);
+  if (dest == NULL) {
+    return NULL;
+  }
+  memcmp(dest, source, dest_size);
+  dest[dest_size] = '\0';
+  return dest;
+}
```

Comment:
```
#if !defined(HAVE_STRDUP) || (HAVE_STRDUP == 0) /* Fallback */
#include <stdlib.h>

char *strndup(...) {
  // ...
}
#endif
```

so you can make strndup directly, then we can revert some changes of `elf-create-argp.c`

Reply:
I've made the changes in f37a0c2. Since the idea of a wrapper function was dropped, I simplified the conditional to just check `#ifndef HAVE_STRNDUP)`.

Let me know what you think!

Reply:
looks good to me now!